### ISSUE TITLE
Feature gRPC get BB statistics

### ISSUE THREAD

**Issue by mihaigalos** [buildbarn/bb-deployments#29]
This is a placeholder for a feature request for a gRPC "GetStats" channel for getting the statistics for remote execution, remote bazel cache.

For remote cache lookup using Build Barn, one can use the [`--remote_cache_header`](https://docs.bazel.build/versions/master/command-line-reference.html#flag--remote_cache_header) for a cache get/put/update. The "GetStats" could take the key used on the cache operation to get a statistic on how well the cache is performing (hit rate, hit rate %, etc.).

**Comment by EdSchouten** [comment]
Are you aware that all Buildbarn components expose a huge number of Prometheus metrics? Just point your browser to /metrics on the HTTP endpoint.

Are there any statistics there that you feel are missing?

**Comment by mihaigalos** [comment]
Yes, I am aware. We could not find any relevant data by consuming Prometheus data in Grafana.
The only alternative would be to have BB produce logs which we would then post-process. I like the gRPC approach better.

**Comment by mickael-carl** [comment]
Which metrics would you be missing?

**Comment by mihaigalos** [comment]
Let me start over, sorry for not making things clear in the first place.

We are using BB also as a bazel remote cache. bazel communicates with the frontend and fetches data from a scalable storage backend, which we can scale as we need.

The problem is, there is no way to ask BB how well it is performing / how many hits in the cache it has.
Of course we can do this on the bazel client side, but a better approach would be to have it centralized over at BB.

Here's the problem: BB doesn't know which cache entry belongs to which client. Clients may be individual developers and CI jobs.

By using the  [`--remote_cache_header`](https://docs.bazel.build/versions/master/command-line-reference.html#flag--remote_cache_header), one can specify a (key,value) which might hold some info as to who the client is. Example: `--remote_cache_header=ci_job=foo` or `--remote_cache_header=dev=mihai`.

Essentially, I would like to be able to query BB for the hit rate for the `foo` CI buildjob or the user `mihai`. I want to be able to ideally export that data to Prometheus.
